clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackInstance.groovy

```diff
@@ -16,64 +16,120 @@
 
 package com.netflix.spinnaker.clouddriver.openstack.model
 
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.databind.ObjectMapper
 import com.netflix.spinnaker.clouddriver.model.HealthState
 import com.netflix.spinnaker.clouddriver.model.Instance
+import com.netflix.spinnaker.clouddriver.openstack.provider.OpenstackInfrastructureProvider
 import groovy.transform.Canonical
+import org.openstack4j.model.compute.Address
 import org.openstack4j.model.compute.Server
 
-import static org.openstack4j.model.compute.Server.Status
-
 @Canonical
-class OpenstackInstance implements Instance, Serializable {
+class OpenstackInstance {
   String name
   String instanceId
   Long launchTime
   String zone
   String region
-  String status
   String keyName
   Map<String, String> metadata
   String account
-
-  //TODO - Determine if load balancers, security groups, and server groups are needed
-  // TODO - Determine which external health checks matter
-  @Override
-  List<Map<String, String>> getHealth() {
-    [[
-       state      : healthState.toString(),
-       zone       : zone,
-       type       : 'serverGroup',
-       description: ''
-     ]]
-  }
-
-  //TODO - Further define health states ... There are 18 OP and 7 spinnaker states.
-  @Override
-  HealthState getHealthState() {
-    switch (Status.forValue(status)) {
-      case Status.ACTIVE:
-        HealthState.Up
-        break
-      case Status.BUILD:
-        HealthState.Starting
-        break
-      case Status.UNKNOWN:
-        HealthState.Unknown
-        break
-      default:
-        HealthState.Down
-    }
-  }
+  String ipv6
+  List<OpenstackLoadBalancerHealth> loadBalancerHealths = []
+  OpenstackInstanceHealth instanceHealth
 
   static OpenstackInstance from(Server server, String account, String region) {
+    Address fixedIpAddress = server?.addresses?.getAddresses('private')?.find { it.type == 'fixed' && it.version == 6 }
```

Comment:
Helping to correlate instances with members (which only have ips, not instance ids)